src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/directives/pipeline_setup.js

```diff
@@ -0,0 +1,87 @@
+import uiModules from 'ui/modules';
+import _ from 'lodash';
+import Pipeline from '../lib/pipeline';
+import Processor from '../lib/processor';
+import angular from 'angular';
+import processorTypes from '../lib/processor_types';
+import savedPipeline from '../sample_pipeline.json'; //temp for debugging purposes
+
+require('../services/ingest');
+require('../styles/_pipeline_setup.less');
+require('./pipeline_output');
+require('./source_data');
+require('./processors');
+
+const app = uiModules.get('kibana');
+
+app.directive('pipelineSetup', function () {
+  return {
+    restrict: 'E',
+    template: require('../views/pipeline_setup.html'),
+    scope: {
+      samples: '=',
+      pipeline: '='
+    },
+    controller: function ($scope, ingest, debounce, Notifier) {
+      const notify = new Notifier({ location: `Ingest Pipeline Setup` });
+      $scope.processorTypes = _.sortBy(processorTypes, 'title');
+      $scope.sample = {};
+
+      const pipeline = new Pipeline(processorTypes, Processor);
+      if ($scope.pipeline) {
+        pipeline.load($scope.pipeline);
+        $scope.sample = $scope.pipeline.input;
+      }
+      $scope.pipeline = pipeline;
```

Comment:
What is the purpose of lines 30-35? I'm guessing it has something to do with going back and forth between wizard steps, but an inline comment would be helpful.